src/H5private.h

```diff
@@ -1014,7 +1056,14 @@ H5_DLL herr_t H5_trace_args(struct H5RS_str_t *rs, const char *type, va_list ap)
 /* global library version information string */
 extern char H5_lib_vers_info_g[];
 
-#ifdef H5_HAVE_THREADSAFE
+/* Both the 'threadsafe' and 'concurrency' options provide threadsafely for
+ * API calls.
+ */
+#if defined(H5_HAVE_THREADSAFE) || defined(H5_HAVE_CONCURRENCY)
```

Comment:
Set H5_HAVE_THREADSAFE_API if either of the macros that indicate threadsafe APIs are set.